pkg/odo/genericclioptions/resolve.go

```diff
@@ -27,15 +27,12 @@ func (o *internalCxt) resolveProjectAndNamespace(cmdline cmdline.Cmdline, config
 		}
 		namespace = projectFlag
 	} else {
-		namespace = configProvider.GetNamespace()
-		if namespace == "" {
-			namespace = o.KClient.GetCurrentNamespace()
-			if len(namespace) <= 0 {
-				errFormat := "Could not get current namespace. Please create or set a namespace\n"
-				err := checkProjectCreateOrDeleteOnlyOnInvalidNamespace(cmdline, errFormat)
-				if err != nil {
-					return err
-				}
+		namespace = o.KClient.GetCurrentNamespace()
+		if len(namespace) <= 0 {
+			errFormat := "Could not get current namespace. Please create or set a namespace\n"
```

Comment:
This might be out of scope for this PR, but we need to streamline namespace vs project throughout all the UX messages. We should use project if it's OCP and namespace if K8s.

Reply:
I prefer to keep this for a UX-relative PR